Fix with base gcc:

  up-device.c:277: error: #pragma GCC diagnostic not allowed inside functions

Index: libupower-glib/up-device.c
--- libupower-glib/up-device.c.orig
+++ libupower-glib/up-device.c
@@ -279,10 +279,7 @@ up_device_to_text (UpDevice *device)
 	/* get a human readable time */
 	t = (time_t) up_exported_device_get_update_time (priv->proxy_device);
 	time_tm = localtime (&t);
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wformat-y2k"
 	strftime (time_buf, sizeof time_buf, "%c", time_tm);
-#pragma GCC diagnostic pop
 
 	string = g_string_new ("");
 	if (!is_display)
